### PR TITLE
eclipses: 2020-06 -> 2020-09

### PR DESCRIPTION
**Pull request by jerith666**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc @rycee

### CODE REVIEW

File: pkgs/applications/editors/eclipse/default.nix [hunk 1]
```diff
@@ -10,13 +10,16 @@ assert stdenv ? glibc;
 
 # https://download.eclipse.org/eclipse/downloads/ is the main place to
 # find the downloads needed for new versions
+#
+# to test:
+# $ for e in cpp modeling platform sdk java committers; do nix build -f default.nix pkgs.eclipses.${e} -o eclipses.${e}; done
```
Comment (rycee) [inline pkgs/applications/editors/eclipse/default.nix:15]: I think a `eclipse-` is missing here:
```suggestion
# $ for e in cpp modeling platform sdk java committers; do nix build -f . pkgs.eclipses.eclipse-${e} -o eclipses.${e}; done
(also default.nix could be simplified to .)
```

Comment (jerith666) [reply pkgs/applications/editors/eclipse/default.nix]: Um ... it was right in my `.bash_history`?  Thanks!  :facepalm:

**Review by mohe2015 — APPROVED** [review]
I tested eclipse-java and it properly upgraded my workspace and running / opening / editing projects still works properly.

**Comment by rycee** [comment]
Thanks to @jerith666 and @mohe2015, much appreciated. Rebased to master in a7cad45d933370424fd12d3a1d46ac88599dba63.

**Pull request closed without merging**